test_handler.go

```diff
@@ -73,10 +83,33 @@ func testHandler(w http.ResponseWriter, r *http.Request) {
         http.Error(w, err.Error(), http.StatusInternalServerError)
         return
     }
-    testRunsJSON := string(testRunsBytes)
 
-    if err := templates.ExecuteTemplate(w, "index.html", testRunsJSON); err != nil {
+    data := struct {
+        TestRuns string
+        SHA      string
+    }{
+        string(testRunsBytes),
+        runSHA,
+    }
+
+    if err := templates.ExecuteTemplate(w, "index.html", data); err != nil {
         http.Error(w, err.Error(), http.StatusInternalServerError)
         return
     }
 }
+
+func getRunSHA(r *http.Request) (runSHA string, err error) {
+    // Get the SHA for the run being loaded (the first part of the path.)
+    runSHA = "latest"
+    params, err := url.ParseQuery(r.URL.RawQuery)
+    if err != nil {
+        return
+    }
+
+    runParam := params.Get("sha")
+    regex := regexp.MustCompile("[0-f]{10}")
```

Comment:
Cool, I didn't know you could do `0-f`!

Reply:
Actually, you can't (includes a wider range of chars). It would be cool if you could, but my bad.